src/features/filter.ts

```diff
@@ -1,12 +1,25 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Status } from '../types/Status';
 
-const initialState = {
+type Filter = {
+  query: string;
+  status: Status;
+};
+
+const initialState: Filter = {
   query: '',
   status: 'all',
 };
 
 export const filterSlice = createSlice({
   name: 'filter',
   initialState,
-  reducers: {},
+  reducers: {
+    setStatus(filter: Filter, { payload }: PayloadAction<Status>): Filter {
```

Comment:
The reducer function `setStatus` should have the state as the first parameter and the action as the second parameter. Correct the function signature to: `setStatus(state: Filter, action: PayloadAction<Status>): Filter { return { ...state, status: action.payload }; }`.